modules/Netbox.py

```diff
@@ -0,0 +1,64 @@
+import pynetbox
+
+
+class Netbox:
+    def __init__(self, nb_config):
+        self.nb_config = nb_config
+        self.nb = self._login()
+
+    def _login(self):
+        """
+        Login to netbox
+        :return:
+        """
+        nb = pynetbox.api(url=self.nb_config["netbox_url"], token=self.nb_config["netbox_token"])
+        return nb
+
+    def get_all_av_zones_from_region(self, region):
+        """
+        Get all availability zones from region.
+        :param region: region name
+        :return: List of availability zones
+        """
+        av_zones = self.nb.dcim.sites.filter(region=region)
+        return av_zones
+
+    def get_all_bb_from_av_zones(self, av_zones):
+        """
+        Get all building blocks from availability zones.
+        :param av_zone:
+        :return: list of bb's from availability zone
+        """
+        bbs = {}
+        for av_zone in av_zones:
+            ucs_servers = self.nb.dcim.devices.filter(q=self.nb_config["query"], tenant=self.nb_config["tenant"],
+                                                      status=self.nb_config["status"], site=av_zone)
+            bbs.setdefault(av_zone, []).extend([ucs.name.split("-")[1].lower() for ucs in ucs_servers])
```

Comment:
first I struggled with this syntax, but figured out, it is way smarter to do it this way than setting the keys traditionally. Thanks :)

Reply:
Thanks